pkg/resource/deploy/step_generator.go

```diff
@@ -296,9 +330,9 @@ func (sg *stepGenerator) GenerateSteps(event RegisterResourceEvent) ([]Step, err
 					var failures []plugin.CheckFailure
 					inputs, failures, err = prov.Check(urn, nil, goal.Properties, allowUnknowns)
 					if err != nil {
-						return nil, err
+						return nil, Undefined, err
 					} else if sg.issueCheckErrors(new, urn, failures) {
-						return nil, errors.New("One or more resource validation errors occurred; refusing to proceed")
+						return nil, Bail, nil
```

Comment:
expected change?

Reply:
Yeah - I missed this one earlier. Sorry, I thought this made it into the first commit.